### PR TITLE
allow latest versions of phpstan

### LINKED ISSUE #6428: phpstan is out of date
Is there a reason to pin phpstan dependency to patch version?
If this can't be changed, can you please update phpstan/phpstan to 0.12.88?

### PR DESCRIPTION
**Pull request by garak**
Fix #6428

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     },
     "conflict": {
         "phpstan/phpdoc-parser": "<=0.5.3",
-        "phpstan/phpstan": "<=0.12.82",
+        "phpstan/phpstan": "<=0.12.88",
```
Comment (TomasVotruba) [inline composer.json:24]: This seems a bit confusing.
How can composer require 0.12.86 and conflict with 0.12.88 and lower at the same time?

Comment (garak) [reply composer.json:24]: my bad, I looked in the wrong section

**Comment by samsonasik** [comment]
@garak please send PR to https://github.com/rectorphp/rector-src instead. You can look the composer.json on https://github.com/rectorphp/rector-src/blob/main/build/target-repository/composer.json

**Comment by garak** [comment]
> @garak please send PR to https://github.com/rectorphp/rector-src instead. You can look the composer.json on https://github.com/rectorphp/rector-src/blob/main/build/target-repository/composer.json

Maybe this is worth a mention in contributing file?

**Pull request closed without merging**